EIPS/eip-draft_restrict_precompile_address_range.md

```diff
@@ -0,0 +1,39 @@
+---
+eip: 1352
+title: Specify restricted address range for precompiles/system contracts
+author: Alex Beregszaszi (axic)
+discussions-to: https://ethereum-magicians.org/t/eip-1352-specify-restricted-address-range-for-precompiles-system-contracts/1151
+status: Draft
+type: Standards
+category: Core
+created: 2018-07-27
+---
+
+## Simple Summary
+Specify an Ethereum address range occupied by precompiles and future system contracts. Regular accounts and contracts cannot obtain such an address.
+
+## Abstract
+The address range between 0x0000000000000000000000000000000000000000 and 0x00000000000000000000000000000000000000ff is reserved for precompiles and system contracts.
+
+## Motivation
+This will simplify certain future features where unless this is implemented, several exceptions must be specified.
```

Comment:
Can you expand on this? What will it simplify?

Reply:
Examples:
- reducing call cost for precompiles
- "future system contracts" can be specified much simpler if they can refer to "precompiles" without the need to specify all their addresses (which is a moving target)

Reply:
I actually had the idea of splitting it into two:
- 0x0 to 0xff for precompiles
- 0x100 to 0x1ff for "system contracts", which have code in the state (e.g. blockhash refactoring or (the now obsolete) casper contract).

Reply:
I would like to make a distinction between precompiles (no code in state) and system contracts (code in state). What ranges to you propose for these two?

Reply:
@axic considering that the definition of a "precompile" may be revised a bit if we introduce wasm precompiles that will likely be more numerous, then i would say that the reserved address range should be at least 65536 addresses long.

This also depends on whether we want to reserve addresses for wasm precompiles, or just consider them a different class of contract in the state.

Reply:
Changed the reserved range.